agent/local/state_test.go

```diff
@@ -373,36 +373,41 @@ func TestAgentAntiEntropy_Services_ConnectProxy(t *testing.T) {
 	// We should have 5 services (consul included)
 	assert.Len(services.NodeServices.Services, 5)
 
-	// All the services should match
+	// Check that virtual IPs have been set
 	vips := make(map[string]struct{})
-	srv1.TaggedAddresses = nil
-	srv2.TaggedAddresses = nil
-	for id, serv := range services.NodeServices.Services {
-		serv.CreateIndex, serv.ModifyIndex = 0, 0
+	for _, serv := range services.NodeServices.Services {
 		if serv.TaggedAddresses != nil {
 			serviceVIP := serv.TaggedAddresses[structs.TaggedAddressVirtualIP].Address
 			assert.NotEmpty(serviceVIP)
 			vips[serviceVIP] = struct{}{}
 		}
-		serv.TaggedAddresses = nil
```

Comment:
Originally we nil-ed `TaggedAddresses` so the nondeterministic values didn't fail.
I think it's more correct to sync local with remote and retry until they pass